staging/src/k8s.io/client-go/transport/spdy/spdy.go

```diff
@@ -20,12 +20,16 @@ import (
 	"fmt"
 	"net/http"
 	"net/url"
+	"time"
 
 	"k8s.io/apimachinery/pkg/util/httpstream"
 	"k8s.io/apimachinery/pkg/util/httpstream/spdy"
 	restclient "k8s.io/client-go/rest"
 )
 
+// TODO: should we make the period configurable?
+const pingPeriod = time.Second * 20
```

Comment:
The 20 second is based on the bug report https://github.com/kubernetes/kubernetes/issues/94301. People might need a different pingPeriod to satisfy different load balancers.

I think we should expose this variable as a flag of `kubectl exec` and `kubectl port-forward` commands. That will need approvals from @kubernetes/sig-cli-api-reviews.

I'm ok with hard-code it to 20 seconds for now and make it configurable in a follow-up.

Can you remove the TODO?

Reply:
Given the [example config for a loadbalancer](https://github.com/kubernetes/kubeadm/blob/master/docs/ha-considerations.md#haproxy-configuration) sets a timeout to 20 seconds having the ping every 20 seconds will most likely cause problems "in the wild". I see this simply as cutting it too close.

I would really prefer a ping interval of something like 5 or 10 seconds because of that.

Reply:
I vote let's hard code to 5s and not make it configurable.

Reply:
I have updated the ping period to 5s.